Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -289,7 +334,14 @@ extension NavigationMapView {
                     }
                     
                     var hideWayName = true
-                    if smallestLabelDistance < 5 {
+                    if latestFeature != similarFeature {
+                        let style = self.navigationMapView.mapView.mapboxMap.style
+                        if let similarFeature = similarFeature,
+                           self.navigationView.wayNameView.setupWith(roadFeature: similarFeature,
```

Comment:
While you’re here, I didn’t realize this method was publicly using a non-Swift type:

https://github.com/mapbox/mapbox-navigation-ios/blob/83eaeba5d6421a36193f9caae403685afd1b0bf6/Sources/MapboxNavigation/WayNameView.swift#L70

We should rename it to `update(feature:style:)`, and the `feature` argument should be of type Feature (from Turf), not MBXFeature. Hopefully that’s a small enough change that it can happen as part of this PR, but feel free to break it out separately if it turns into something larger.

Reply:
Thanks for the suggestion. It's included in #3039 .

Reply:
Thanks, I made the same request in https://github.com/mapbox/mapbox-navigation-ios/pull/3039#discussion_r641789971. (`MapboxCommon.Feature` is not the same as `Turf.Feature`.)